res/controllers/Numark-MixstreamPro-scripts.js

```diff
@@ -0,0 +1,1604 @@
+    
+    return 
+};
+
+MixstreamPro.JogMSB_1 = function(channel, control, value, status, group) {
+        let MSB = value;
+        let POS = engine.getValue(group, "playposition")
+        let deckNumber = script.deckFromGroup(group);
+
+        switch(true) {
+            case POS <= 0:
+                 engine.setValue(group, "playposition", 1)
+              break;
+            case POS >= 1 :
+                 engine.setValue(group, "playposition", 0)
+              break;
+          }
+
+        if (engine.isScratching(deckNumber)) {
+            if (MSB >= MixstreamPro.previousJogValue1) {
+                engine.scratchTick(deckNumber, MSB/2);  
+                MixstreamPro.previousJogValue1 = value;
+            }
+            else 
+                {
+                engine.scratchTick(deckNumber, -MSB/2);  
+                MixstreamPro.previousJogValue1 = value;
+                }
+        } 
+        else  
+        {
+            if (MSB >= MixstreamPro.previousJogValue1) {
+                engine.setValue(group, "jog", MSB * MixstreamPro.jogSensitivity);
+                MixstreamPro.previousJogValue1 = value;
+            }
+        else 
+            {
+            engine.setValue(group, "jog", -MSB * MixstreamPro.jogSensitivity);
+            MixstreamPro.previousJogValue1 = value;
+            }   
+        }
+};
+
+ 
+MixstreamPro.JogLSB_2 = function(channel, control, value, status, group) {
+     
+     return
+ };
+ 
+ MixstreamPro.JogMSB_2 = function(channel, control, value, status, group) {
+        let MSB = value;
+        let POS = engine.getValue(group, "playposition")
+        let deckNumber = script.deckFromGroup(group);
+
+        switch(true) {
+            case POS <= 0:
+                 engine.setValue(group, "playposition", 1)
+              break;
+            case POS >= 1 :
+                 engine.setValue(group, "playposition", 0)
+              break;
+          }
+
+        if (engine.isScratching(deckNumber)) {
+            if (MSB >= MixstreamPro.previousJogValue1) {
+                engine.scratchTick(deckNumber, MSB/2);  
+                MixstreamPro.previousJogValue1 = value;
+            }
+        else 
+            {
+            engine.scratchTick(deckNumber, -MSB/2);  
+            MixstreamPro.previousJogValue1 = value;
+            }
+        } 
+        else  
+        {
+            if (MSB >= MixstreamPro.previousJogValue1) {
+                engine.setValue(group, "jog", MSB * MixstreamPro.jogSensitivity);
+                MixstreamPro.previousJogValue1 = value;
+            }
+        else 
+            {
+            engine.setValue(group, "jog", -MSB * MixstreamPro.jogSensitivity);
+            MixstreamPro.previousJogValue1 = value;
+            }   
+        }
+ };
+
+ 
+MixstreamPro.blinktimer4 = 0
+MixstreamPro.hotcuevalue1 = 0
+MixstreamPro.LEDblink4 = true
+ 
+MixstreamPro.track_loaded1 = function(channel, control, value, status, group) {

```

Comment:
Is this a callback function?
If no, how is it called?

Same here, refactor to one function and use `group`